### ISSUE TITLE
Connection failed: Not_found

### ISSUE THREAD

**Issue by Creepercdn** [savonet/liquidsoap#1372]
**Describe the bug**
```
2020/10/03 23:25:41 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:25:41 [main:2] Connection failed: Not_found
2020/10/03 23:25:41 [main:3] Will try again in 3.00 sec.
2020/10/03 23:25:45 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:25:45 [main:2] Connection failed: Not_found
2020/10/03 23:25:45 [main:3] Will try again in 3.00 sec.
2020/10/03 23:25:49 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:25:49 [main:2] Connection failed: Not_found
2020/10/03 23:25:49 [main:3] Will try again in 3.00 sec.
2020/10/03 23:25:53 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:25:53 [main:2] Connection failed: Not_found
2020/10/03 23:25:53 [main:3] Will try again in 3.00 sec.
2020/10/03 23:25:57 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:25:57 [main:2] Connection failed: Not_found
2020/10/03 23:25:57 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:01 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:01 [main:2] Connection failed: Not_found
2020/10/03 23:26:01 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:05 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:05 [main:2] Connection failed: Not_found
2020/10/03 23:26:05 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:09 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:09 [main:2] Connection failed: Not_found
2020/10/03 23:26:09 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:13 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:13 [main:2] Connection failed: Not_found
2020/10/03 23:26:13 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:17 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:17 [main:2] Connection failed: Not_found
2020/10/03 23:26:17 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:21 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:21 [main:2] Connection failed: Not_found
2020/10/03 23:26:21 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:25 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:25 [main:2] Connection failed: Not_found
2020/10/03 23:26:25 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:29 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:29 [main:2] Connection failed: Not_found
2020/10/03 23:26:29 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:33 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:33 [main:2] Connection failed: Not_found
2020/10/03 23:26:33 [main:3] Will try again in 3.00 sec.
2020/10/03 23:26:37 [main:3] Connecting mount main for source@localhost...
2020/10/03 23:26:37 [main:2] Connection failed: Not_found
2020/10/03 23:26:37 [main:3] Will try again in 3.00 sec.
```

**To Reproduce**
```
output.icecast(%vorbis,
    host="127.0.0.1",port=8000,
    password="hackme",
    mount=id,
    mksafe(delay(5.0,radio))
)
```

**Expected behavior**

**Version details**
 - OS: Microsoft Windows Server 1607
 - Version 1.4.3

**Install method**
GitHub Releases

**Comment by toots** [comment]
Hi! Do you have an icecast server running locally?

**Comment by Creepercdn** [comment]
Yes, but this error is only thrown when running as a Python sub process.

**Comment by toots** [comment]
What do you mean?

**Comment by Creepercdn** [comment]
Do you know "subprocess" module in Python? It can running a process as Python process's sub process.

**Comment by Creepercdn** [comment]
So I want use this module to manager liquidsoap process. I want to auto start multiple stream on different process.

**Comment by toots** [comment]
Ho interesting. We should be able to generate a trace of the exception by setting the following:
```
def on_error (_) =
  (-1.)
end

output.icecast(%vorbis,
    host="127.0.0.1",port=8000,
    password="hackme",
    mount=id,
   on_error=on_error,
    mksafe(delay(5.0,radio))
)
```

**Comment by Creepercdn** [comment]
OK. This is log:
```
2020/10/16 22:21:16 >>> LOG START
2020/10/16 22:21:16 [main:3] Liquidsoap 1.4.3
2020/10/16 22:21:16 [main:3] Using: bytes=[distributed with OCaml 4.02 or above] pcre=7.4.1 sedlex=2.2 menhirLib=20181113 dtools=0.4.0 duppy=0.8.0 cry=0.6.4 mm=0.5.0 xmlplaylist=0.1.4 lastfm=0.3.2 ogg=0.5.2 vorbis=0.7.1 opus=0.1.3 speex=0.2.1 mad=0.4.6 flac=0.1.7 flac.ogg=0.1.7 dynlink=[distributed with Ocaml] lame=0.3.3 gstreamer=0.3.0 fdkaac=0.3.1 theora=0.3.1 ao=0.2.1 samplerate=0.1.4 taglib=0.3.5 ssl=0.5.9 camomile=1.0.1 yojson=1.7.0 faad=0.4.0 portaudio=0.2.1 winsvc=0.1.1
2020/10/16 22:21:16 [gstreamer.loader:3] Loaded GStreamer 1.16.0 0
2020/10/16 22:21:16 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz master.
2020/10/16 22:21:16 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2020/10/16 22:21:16 [frame:3] Targetting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2020/10/16 22:21:16 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2020/10/16 22:21:16 [decoder:3] Method "OGG" accepted "C:/Users/Administrator/Desktop/scex_radio/radio/default.ogg".
2020/10/16 22:21:16 [sandbox:3] Could not find binary bwrap, disabling sandboxing..
2020/10/16 22:21:16 [video.converter:3] Couldn't find preferred video converter: gavl.
2020/10/16 22:21:16 [audio.converter:3] Using samplerate converter: libsamplerate.
2020/10/16 22:21:16 [bellad(dot)m3u8:3] Loading playlist...
2020/10/16 22:21:16 [bellad(dot)m3u8:3] No mime type specified, trying autodetection.
2020/10/16 22:21:16 [bellad(dot)m3u8:3] Playlist treated as format application/x-mpegURL
2020/10/16 22:21:16 [bellad(dot)m3u8:3] Successfully loaded a playlist of 1 tracks.
2020/10/16 22:21:16 [main(dot)m3u8:3] Loading playlist...
2020/10/16 22:21:16 [main(dot)m3u8:3] No mime type specified, trying autodetection.
2020/10/16 22:21:16 [main(dot)m3u8:3] Playlist treated as format application/x-mpegURL
2020/10/16 22:21:16 [main(dot)m3u8:3] Successfully loaded a playlist of 10 tracks.
2020/10/16 22:21:16 [single_8412:3] "C:/Users/Administrator/Desktop/scex_radio/radio/default.ogg" is static, resolving once for all...
2020/10/16 22:21:16 [single_8412:3] Prepared "C:/Users/Administrator/Desktop/scex_radio/radio/default.ogg" (RID 0).
2020/10/16 22:21:16 [main:3] Connecting mount main for source@127.0.0.1...
2020/10/16 22:21:16 [procol.external:3] Failed to fetch mime-type for http://49.233.66.25/media/audio/home%20in%20db.ogg.
2020/10/16 22:21:16 [procol.external:3] No known file extension for mime: 
2020/10/16 22:21:16 [decoder:3] Method "OGG" accepted "C:/Users/Administrator/Desktop/scex_radio/radio\\audio/bel.ogg".
2020/10/16 22:21:16 [main:2] Connection failed: Not_found
2020/10/16 22:21:16 [clock.wallclock_main:2] Error when starting output main: Not_found!
2020/10/16 22:21:16 [clock.wallclock_main:3] Raised by primitive operation at file "src/pcre.ml", line 723, characters 8-94
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "src/pcre.ml", line 723, characters 8-94
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "cry.ml", line 574, characters 2-27
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "cry.ml", line 696, characters 19-43
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "outputs/icecast2.ml", line 581, characters 8-37
2020/10/16 22:21:16 [clock.wallclock_main:3] Re-raised at file "outputs/icecast2.ml", line 596, characters 13-20
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "tools/start_stop.ml", line 108, characters 8-18
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "clock.ml", line 245, characters 20-38
2020/10/16 22:21:16 [lang:3] Failed to execute curl -sL "****" -o ".\\liq-processac92ba.osb": ("exit", "2")
2020/10/16 22:21:16 [main:3] Shutdown started!
2020/10/16 22:21:16 [main:3] Waiting for main threads to terminate...
2020/10/16 22:21:16 [main:3] Threads terminated.
2020/10/16 22:21:16 [threads:3] Shutting down scheduler...
2020/10/16 22:21:16 [threads:3] Scheduler shut down.
2020/10/16 22:21:16 [threads:3] Waiting for queue threads to terminate...
2020/10/16 22:21:16 [threads:3] Queues shut down
2020/10/16 22:21:16 [main:3] Cleaning downloaded files...
2020/10/16 22:21:16 [main:3] Freeing memory...
2020/10/16 22:21:17 >>> LOG END
```

**Comment by Creepercdn** [comment]
2020/10/16 22:21:16 >>> LOG START
2020/10/16 22:21:16 [main:3] Liquidsoap 1.4.3
2020/10/16 22:21:16 [main:3] Using: bytes=[distributed with OCaml 4.02 or above] pcre=7.4.1 sedlex=2.2 menhirLib=20181113 dtools=0.4.0 duppy=0.8.0 cry=0.6.4 mm=0.5.0 xmlplaylist=0.1.4 lastfm=0.3.2 ogg=0.5.2 vorbis=0.7.1 opus=0.1.3 speex=0.2.1 mad=0.4.6 flac=0.1.7 flac.ogg=0.1.7 dynlink=[distributed with Ocaml] lame=0.3.3 gstreamer=0.3.0 fdkaac=0.3.1 theora=0.3.1 ao=0.2.1 samplerate=0.1.4 taglib=0.3.5 ssl=0.5.9 camomile=1.0.1 yojson=1.7.0 faad=0.4.0 portaudio=0.2.1 winsvc=0.1.1
2020/10/16 22:21:16 [gstreamer.loader:3] Loaded GStreamer 1.16.0 0
2020/10/16 22:21:16 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz master.
2020/10/16 22:21:16 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2020/10/16 22:21:16 [frame:3] Targetting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2020/10/16 22:21:16 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2020/10/16 22:21:16 [decoder:3] Method "OGG" accepted "***".
2020/10/16 22:21:16 [sandbox:3] Could not find binary bwrap, disabling sandboxing..
2020/10/16 22:21:16 [video.converter:3] Couldn't find preferred video converter: gavl.
2020/10/16 22:21:16 [audio.converter:3] Using samplerate converter: libsamplerate.
2020/10/16 22:21:16 [bellad(dot)m3u8:3] Loading playlist...
2020/10/16 22:21:16 [bellad(dot)m3u8:3] No mime type specified, trying autodetection.
2020/10/16 22:21:16 [bellad(dot)m3u8:3] Playlist treated as format application/x-mpegURL
2020/10/16 22:21:16 [bellad(dot)m3u8:3] Successfully loaded a playlist of 1 tracks.
2020/10/16 22:21:16 [main(dot)m3u8:3] Loading playlist...
2020/10/16 22:21:16 [main(dot)m3u8:3] No mime type specified, trying autodetection.
2020/10/16 22:21:16 [main(dot)m3u8:3] Playlist treated as format application/x-mpegURL
2020/10/16 22:21:16 [main(dot)m3u8:3] Successfully loaded a playlist of 10 tracks.
2020/10/16 22:21:16 [single_8412:3] "***" is static, resolving once for all...
2020/10/16 22:21:16 [single_8412:3] Prepared "***" (RID 0).
2020/10/16 22:21:16 [main:3] Connecting mount main for source@127.0.0.1...
2020/10/16 22:21:16 [procol.external:3] Failed to fetch mime-type for ***.
2020/10/16 22:21:16 [procol.external:3] No known file extension for mime: 
2020/10/16 22:21:16 [decoder:3] Method "OGG" accepted "***".
2020/10/16 22:21:16 [main:2] Connection failed: Not_found
2020/10/16 22:21:16 [clock.wallclock_main:2] Error when starting output main: Not_found!
2020/10/16 22:21:16 [clock.wallclock_main:3] Raised by primitive operation at file "src/pcre.ml", line 723, characters 8-94
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "src/pcre.ml", line 723, characters 8-94
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "cry.ml", line 574, characters 2-27
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "cry.ml", line 696, characters 19-43
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "outputs/icecast2.ml", line 581, characters 8-37
2020/10/16 22:21:16 [clock.wallclock_main:3] Re-raised at file "outputs/icecast2.ml", line 596, characters 13-20
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "tools/start_stop.ml", line 108, characters 8-18
2020/10/16 22:21:16 [clock.wallclock_main:3] Called from file "clock.ml", line 245, characters 20-38
2020/10/16 22:21:16 [lang:3] Failed to execute curl -sL "***" -o ".\\liq-processac92ba.osb": ("exit", "2")
2020/10/16 22:21:16 [main:3] Shutdown started!
2020/10/16 22:21:16 [main:3] Waiting for main threads to terminate...
2020/10/16 22:21:16 [main:3] Threads terminated.
2020/10/16 22:21:16 [threads:3] Shutting down scheduler...
2020/10/16 22:21:16 [threads:3] Scheduler shut down.
2020/10/16 22:21:16 [threads:3] Waiting for queue threads to terminate...
2020/10/16 22:21:16 [threads:3] Queues shut down
2020/10/16 22:21:16 [main:3] Cleaning downloaded files...
2020/10/16 22:21:16 [main:3] Freeing memory...
2020/10/16 22:21:17 >>> LOG END

**Comment by toots** [comment]
This is very confusing. `Cry` does not call or depend on `Pcre`, I have no idea why the two are showing up in this stacktrace..

**Comment by toots** [comment]
Ok, I _think_ that I found the source for this one. I believe it was due to a wrong way of reporting exception after it was raised. This should be fixed in 68b13f2 and should show up shortly at: https://github.com/savonet/liquidsoap/releases/tag/v1.4.4-pre-release. It would be nice if you had a second to confirm that the exception is properly reported with the latest build there. You can track the status of the build here: https://github.com/savonet/liquidsoap/actions/runs/454690439. Thanks for reporting!
